docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -124,453 +122,477 @@ This article walks you through the process of enabling single sign-on (SSO) in a
+    // there will be a code property in the returned object set to a HTTP status (e.g. 401).
+    // Return it to the client. On client side it will get handled in the fail callback of `makeWebServerApiCall`.
+    if (graphData.code) {
+      res.status(500).send({ type: "Microsoft Graph", errorDetails: graphData });
+    } else {
+      // MS Graph data includes OData metadata and eTags that we don't need.
+      // Send only what is actually needed to the client: the item names.
+      const itemNames = [];
+      const oneDriveItems = graphData["value"];
+      for (let item of oneDriveItems) {
+        itemNames.push(item["name"]);
+      }
+
+      res.status(200).send(itemNames);
+    }
     ```
 
-1. Replace `TODO 16` with the following. About this code, note:
-
-    * If Microsoft Graph returns an error, such as invalid or expired token, there will be a code property in the returned object set to a HTTP status (e.g., 401). The code relays the error to the client. It will be caught in the `.fail` callback of `makeGraphApiCall`.
-    * Microsoft Graph data includes OData metadata and eTags that the add-in does not need, so the code constructs a new array containing only the file names to send to the client.
+1. Replace `TODO 15` with the following code. This code specifically checks if the token expired because the client can request a new token and call again.
 
     ```javascript
-    if (graphData.code) {
-        next(createError(graphData.code, "Microsoft Graph error: " + JSON.stringify(graphData)));
-    }
-    else {
-        const itemNames = [];
-        const oneDriveItems = graphData['value'];
-        for (let item of oneDriveItems) {
-            itemNames.push(item['name']);
-        }
-
-        res.send(itemNames)
+    // On rare occasions the SSO access token is unexpired when Office validates it,
+    // but expires by the time it is used in the OBO flow. Microsoft identity platform will respond
+    // with "The provided value for the 'assertion' is not valid. The assertion has expired."
+    // Construct an error message to return to the client so it can refresh the SSO token.
+    if (err.errorMessage.indexOf("AADSTS500133") !== -1) {
+      res.status(500).send({ type: "AADSTS500133", errorDetails: err });
+    } else {
+      res.status(500).send({ type: err.errorCode, errorDetails: err });
     }
     ```
 
-1. Save and close the file.
+The sample must handle both fallback authentication through MSAL and SSO authentication through Office. The sample will try SSO first, and the `authSSO` boolean at the top of the file tracks if the sample is using SSO or has switched to fallback auth.
 
 ## Run the project
 
 1. Ensure that you have some files in your OneDrive so that you can verify the results.
 
 1. Open a command prompt in the root of the `\Begin` folder.
 
-1. Run the command `npm start`.
+1. Run the command `npm install` to install all package dependencies.
```

Comment:
No, but it doesn't hurt if run again. It's here twice in case someone was following the "complete" path and they might have skipped this step earlier. It just saves getting GitHub issues if someone forgot this step.